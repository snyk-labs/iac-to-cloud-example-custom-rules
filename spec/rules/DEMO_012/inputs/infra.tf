resource "github_repository" "valid_by_node_id" {
  name = "valid-by-node-id"
}

resource "github_branch_protection" "valid_by_node_id" {
  repository_id = github_repository.valid_by_node_id.node_id
  pattern       = "main"

  required_pull_request_reviews {
    restrict_dismissals = true
  }
}

resource "github_repository" "valid_by_name" {
  name = "valid-by-name"
}

resource "github_branch_protection" "valid_by_name" {
  repository_id = github_repository.valid_by_name.name
  pattern       = "main"

  required_pull_request_reviews {
    restrict_dismissals = true
  }
}

resource "github_repository" "valid_restrictions" {
  name = "valid-restrictions"
}

resource "github_branch_protection" "valid_restrictions" {
  repository_id = github_repository.valid_restrictions.name
  pattern       = "main"

  required_pull_request_reviews {
    dismissal_restrictions = [
      "snyk/cloud-engines",
    ]
  }
}

resource "github_repository" "invalid" {
  name = "invalid"
}

resource "github_repository" "invalid_implicit" {
  name = "invalid-implicit"
}

resource "github_branch_protection" "invalid_implicit" {
  repository_id = github_repository.invalid_implicit.name
  pattern       = "main"

  required_pull_request_reviews {}
}

resource "github_repository" "invalid_explicit" {
  name = "invalid-explicit"
}

resource "github_branch_protection" "invalid_explicit" {
  repository_id = github_repository.invalid_explicit.name
  pattern       = "main"

  required_pull_request_reviews {
    restrict_dismissals = false
  }
}

resource "github_repository" "invalid_restrictions" {
  name = "invalid-restrictions"
}

resource "github_branch_protection" "invalid_restrictions" {
  repository_id = github_repository.invalid_restrictions.name
  pattern       = "main"

  required_pull_request_reviews {
    dismissal_restrictions = []
  }
}
