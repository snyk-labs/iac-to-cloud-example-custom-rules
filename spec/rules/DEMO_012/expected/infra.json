[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "github_repository.invalid",
    "resource_namespace": "spec/rules/DEMO_012/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_repository.invalid",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 42,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 43,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 42,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": false,
    "ignored": false,
    "resource_id": "github_repository.invalid_explicit",
    "resource_namespace": "spec/rules/DEMO_012/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.invalid_explicit",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 61,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 62,
              "column": 3
            }
          },
          {
            "path": [
              "required_pull_request_reviews",
              0,
              "restrict_dismissals"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 66,
              "column": 5
            }
          }
        ]
      },
      {
        "id": "github_repository.invalid_explicit",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 57,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 58,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 57,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": false,
    "ignored": false,
    "resource_id": "github_repository.invalid_implicit",
    "resource_namespace": "spec/rules/DEMO_012/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.invalid_implicit",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 50,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 51,
              "column": 3
            }
          },
          {
            "path": [
              "required_pull_request_reviews"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 54,
              "column": 3
            }
          }
        ]
      },
      {
        "id": "github_repository.invalid_implicit",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 46,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 47,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 46,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": false,
    "ignored": false,
    "resource_id": "github_repository.invalid_restrictions",
    "resource_namespace": "spec/rules/DEMO_012/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.invalid_restrictions",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 74,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 75,
              "column": 3
            }
          },
          {
            "path": [
              "required_pull_request_reviews",
              0,
              "dismissal_restrictions"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 79,
              "column": 5
            }
          }
        ]
      },
      {
        "id": "github_repository.invalid_restrictions",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 70,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 71,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 70,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "github_repository.valid_by_name",
    "resource_namespace": "spec/rules/DEMO_012/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.valid_by_name",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 18,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 19,
              "column": 3
            }
          },
          {
            "path": [
              "required_pull_request_reviews",
              0,
              "restrict_dismissals"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 23,
              "column": 5
            }
          }
        ]
      },
      {
        "id": "github_repository.valid_by_name",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 14,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 15,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 14,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "github_repository.valid_by_node_id",
    "resource_namespace": "spec/rules/DEMO_012/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.valid_by_node_id",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 5,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 6,
              "column": 3
            }
          },
          {
            "path": [
              "required_pull_request_reviews",
              0,
              "restrict_dismissals"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 10,
              "column": 5
            }
          }
        ]
      },
      {
        "id": "github_repository.valid_by_node_id",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 1,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 2,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 1,
              "column": 1
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "github_repository.valid_restrictions",
    "resource_namespace": "spec/rules/DEMO_012/inputs/infra.tf",
    "resource_type": "github_repository",
    "severity": "low",
    "resources": [
      {
        "id": "github_branch_protection.valid_restrictions",
        "type": "github_branch_protection",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 31,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "repository_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 32,
              "column": 3
            }
          },
          {
            "path": [
              "required_pull_request_reviews",
              0,
              "dismissal_restrictions"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 36,
              "column": 5
            }
          }
        ]
      },
      {
        "id": "github_repository.valid_restrictions",
        "type": "github_repository",
        "namespace": "spec/rules/DEMO_012/inputs/infra.tf",
        "location": [
          {
            "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
            "line": 27,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "name"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 28,
              "column": 3
            }
          },
          {
            "path": [
              "node_id"
            ],
            "location": {
              "filepath": "spec/rules/DEMO_012/inputs/infra.tf",
              "line": 27,
              "column": 1
            }
          }
        ]
      }
    ]
  }
]